lib/split/algorithms/weighted_deterministic.rb

```diff
@@ -0,0 +1,20 @@
+require 'digest/murmurhash'
+module Split
+  module Algorithms
+    module WeightedDeterministic
+      def self.choose_alternative(experiment, split_id)
+        hash = Digest::MurmurHash3_x86_32.rawdigest("#{experiment.key}:#{split_id}")
+
+        weights = experiment.alternatives.map(&:weight)
+        total = weights.inject(:+) * 100
+
+        point = (hash % total)
+
+        experiment.alternatives.zip(weights).each do |n,w|
```

Comment:
hmmm benchmarking doesn't show speed being an issue here. avg execution time for 10k `batch_ab_test` (which uses `choose_alternative`) down from 46.6s to 0.94s. maybe can be more optimized but seems like pretty minor optimization at this point.